source/game.database.bmx

```diff
@@ -235,10 +232,125 @@ Type TDatabaseLoader
 			Notify "Important data is missing:  series:"+totalSeriesCount+"  movies:"+totalMoviesCount+"  news:"+totalNewsCount+"  adcontracts:"+totalContractsCount
 		EndIf
 
+		LoadDatabaseLocalizations(dbDirectory)
 
 		'fix potentially corrupt data
 		FixLoadedData()
 	End Method
+
+
+	Function LoadDatabaseLocalizations(dbDirectory:String)
+		Local langDir:String = dbDirectory+"/lang/"
+		Local dbl:TDatabaseLocalizer = GetDatabaseLocalizer()
+
+		For Local l:TLocalizationLanguage = EachIn TLocalization.languages
+			Local languageId:Int = TLocalization.GetLanguageId(l.languageCode)
+			If languageId < 0 Then Continue
+			Local file:String = langDir + l.languageCode+".xml"
+			If FileType(file) = 1
+				Local xml:TXmlHelper = TXmlHelper.Create(file)
+				Local allGlobalVars:TxmlNode = xml.FindRootChildLC("globalvariables")
+				If allGlobalVars
+					Local gvl:TLocalizationLanguage = dbl.getGlobalVariables(languageId)
+					If Not gvl
+						gvl = new TLocalizationLanguage
+						gvl.languageCode = l.languageCode
+						dbl.globalVariables.insert(languageId, gvl)
+					EndIf
+					Local varName:String
+					Local value:String
+					For Local varNode:TxmlNode = EachIn xml.GetNodeChildElements(allGlobalVars)
+						varName = varNode.getName().toLower()
+						value = varNode.getContent()
+						If varName And value Then gvl.map.insert(varName, value)
+					Next
+				EndIf
+
+				Local NO_VALUE:String = "<NO_VALUE_PRESENT>"
+				Local v:String
+
+				Local nodeAllPersons:TxmlNode
+				nodeAllPersons = xml.FindRootChildLC("persons")
+				Local personsToStore:PersonLocalization[] = new PersonLocalization[0]
+				Local personCollection:TPersonBaseCollection = GetPersonBaseCollection()
+				For Local nodePerson:TxmlNode = EachIn xml.GetNodeChildElements(nodeAllPersons)
+					If nodePerson.getName() <> "person" Then Continue
+					Local data:TData = New TData
+					xml.LoadValuesToData(nodePerson, data, ["guid","first_name", "last_name", "nick_name", "title"])
+					Local guid:String=data.GetString("guid")
+					If guid
+						Local person:TPersonBase = personCollection.GetByGUID(guid)
+						If person
+							Local personToStore:PersonLocalization = new PersonLocalization
+							personToStore.id = person.id
+							v = data.GetString("first_name",NO_VALUE)
```

Comment:
An sich enthält ja "data" nur noch Felder (guid, first_name, ...) die es in der DB laden konnte.
Man koennte also auch ueber deren Eintraege an die Sache rangehen und dann Select case.

Aber was mir hier gerade eher auffaellt ist: `Data.Get(bla, defaultValue)` wuerde (ungetestet) defaultValue zurueckgeben, wenn ein leerer String (aka "Null") hinterlegt wurde.

Man muesste also vorher ueber "Has()" pruefen, ob der Key vorhanden ist (für dieses "get value und gleichzeitig noch ein true/false ob der kex existiert.." hatte ich schon einmal einen Vorschlag fuer die TMaps rausgeschickt, aber evtl. muesste man eine Custom Map anlegen die das bereitstellt)

Reply:
Vielleicht kann man sich den ganzen Zauber überhaupt sparen und davon ausgehen, dass die Spracheinträge die vollständigen Namen enthalten.

Reply:
Oder bei nicht-ueberschreiben den Originalwert reinpacken (der aber dann "ab dann" sich nicht mehr aendern sollte).

Reply:
Ich werde das vereinfachen und das Standard-DB-Schema verwenden. Was nicht angegeben ist, ist ein Leerstring.